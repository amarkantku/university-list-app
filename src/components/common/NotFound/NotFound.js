import React from 'react';
import Layout from '../Layout/Layout';
import { useHistory } from 'react-router-dom';

function NotFound() {
    const history = useHistory();
    return (
        <Layout>
            <div>
                <div className="center">
                <h1>404</h1>
                <h2>UH OH! You're lost.</h2>
                <p>The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage.
                </p>
                <button onClick={() => history.push('/')}> Go to homepage</button>
              </div>
            </div>
        </Layout>
    )
}

export default NotFound;
