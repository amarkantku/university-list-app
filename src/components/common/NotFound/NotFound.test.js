import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import NotFound from './NotFound';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<RouteNotFound />', () => {
  test('component should render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Redirects to correct URL on click', () => {
    const { getByRole } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );
    fireEvent.click(getByRole('button'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});