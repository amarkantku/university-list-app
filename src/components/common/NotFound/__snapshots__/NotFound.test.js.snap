// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteNotFound /> component should render 1`] = `
<DocumentFragment>
  <div
    class="layout row"
    data-testid="layout"
  >
    <header
      data-testid="header"
    >
      <div
        class="header"
      >
        <h1>
          University List App
        </h1>
      </div>
    </header>
    <main
      class="main"
      data-testid="main"
    >
      <h2
        class="heading"
      />
      <div>
        <div
          class="center"
        >
          <h1>
            404
          </h1>
          <h2>
            UH OH! You're lost.
          </h2>
          <p>
            The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage.
          </p>
          <button>
             Go to homepage
          </button>
        </div>
      </div>
    </main>
    <footer
      data-testid="footer"
    >
      <div
        class="footer"
      >
        Footer
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
